compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -287,6 +287,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         expr_ty: Ty<'tcx>,
     ) {
         if let ty::Adt(expected_adt, substs) = expected.kind() {
+            if let hir::ExprKind::Field(base, ident) = expr.kind
+                && let Some(typeck_results) = self.in_progress_typeck_results
+            {
+                let base_ty = typeck_results.borrow().expr_ty(base);
```

Comment:
FnCtxt has `self.typeck_results` which are not `Option<T>`

```suggestion
        if let ty::Adt(expected_adt, substs) = expected.kind() {
            if let hir::ExprKind::Field(base, ident) = expr.kind
            {
                let base_ty = self.typeck_results.borrow().expr_ty(base);
```